pkg/discovery/discovery.go

```diff
@@ -166,6 +170,13 @@ func Run(restConf *rest.Config, cfg *config.Config) (errCount int) {
 		)
 	}
 
+	// Add a log line at the end of the run for clarity. Common problem in timeout situations is that
+	// users do not find the timeout message in the middle of the run logs. Can't just add it with a `defer`
+	// since we'd also like this to appear in the podlogs that get put into the tarball.
+	if tErr, ok := runErr.(timeout); ok && tErr.Timeout() {
```

Comment:
It is a bit of overengineering; but for a long time I've tried to use this approach due to the arguments given here: https://dave.cheney.net/2016/04/27/dont-just-check-errors-handle-them-gracefully

Reply:
Interesting post! Thanks for sharing :) I hadn't considered that approach before. Doesn't apply in this case but I can see how it would make other error handling more flexible as you could have multiple error types implementing a given interface :+1: